blackboxopt/optimizers/testing.py

```diff
@@ -218,9 +237,48 @@ def raises_evaluation_error_when_reporting_unknown_objective(
     return True
 
 
+def respects_fixed_parameter(
+    optimizer_class: Union[
+        Type[SingleObjectiveOptimizer], Type[MultiObjectiveOptimizer]
+    ],
+    optimizer_kwargs: dict,
+):
+    """Check if optimizer's generated evaluation specifications contain the values
+    a parameter in the search space was fixed to.
+
+    Args:
+        optimizer_class: Optimizer to test.
+        optimizer_kwargs: Expected to contain additional arguments for initializing
+            the optimizer. (`search_space` and `objective(s)` are set automatically
+            by the test.)
+
+    Returns:
+        `True` if the test is passed.
+    """
+    space = ps.ParameterSpace()
+    space.add(ps.ContinuousParameter("my_fixed_param", (-10.0, 200.0)))
+
+    fixed_value = 1.0
+    space.fix(my_fixed_param=fixed_value)
+
+    for _ in range(5):
+        opt = _initialize_optimizer(
+            optimizer_class,
+            optimizer_kwargs,
+            objective=Objective("loss", False),
+            objectives=[Objective("loss", False)],
+            space=space,
+        )
+        es = opt.generate_evaluation_specification()
+        assert es.configuration["my_fixed_param"] == fixed_value
```

Comment:
No, since in each iteration the optimizer is created from scratch. Would the optimizer be instantiated outside of the loop, your observation would be accurate.

Reply:
You are absolutely right, but then I have to ask whether we really want to test 5 times if the first proposed eval_spec contains the fixed parameter. Most model-based optimizers would start with a random sample that might come directly from `ParameterSpace.sample()`, rather than building a model. I would argue that it would be more interesting to see if 5 evaluations can be queried if results are reported back. That way, any non-trivial optimizer will be forced to do its non-trivial thing, which is what we probably want to test.

Reply:
Thanks, that makes a lot of sense. I changed it accordingly.